source/MultiGrid/MultiGrid.js

```diff
@@ -0,0 +1,338 @@
+
+    if (
+      fixedRowCount !== nextProps.fixedRowCount ||
+      rowHeight !== nextProps.rowHeight
+    ) {
+      this._topGridHeight = null
+    }
+  }
+
+  render () {
+    const { height, style, width } = this.props
+    const { scrollLeft, scrollTop } = this.state
+
+    return (
+      <div
+        style={{
+          height,
+          width,
+          ...style
+        }}
+      >
+        <div
+          style={{
+            height: this._getTopGridHeight(),
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderTopLeftGrid()}
+          {this._renderTopRightGrid({ scrollLeft })}
+        </div>
+        <div
+          onScroll={this._onScrollVertical}
+          ref={(ref) => {
+            this._scrollingContainer = ref
+          }}
+          style={{
+            height: height - this._getTopGridHeight(),
+            overflowY: 'auto',
+            position: 'relative',
+            width
+          }}
+        >
+          {this._renderBottomLeftGrid({ scrollTop })}
+          {this._renderBottomRightGrid({ scrollTop })}
+        </div>
+      </div>
+    )
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState)
+  }
+
+  _cellRendererBottomLeftGrid ({ rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      rowIndex: rowIndex + fixedRowCount
+    })
+  }
+
+  _cellRendererBottomRightGrid ({ columnIndex, rowIndex, ...rest }) {
+    const { cellRenderer, fixedRowCount } = this.props
+
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + columnIndex,
+      rowIndex: rowIndex + fixedRowCount
+    })
+  }
+
+  _cellRendererTopRightGrid ({ columnIndex, ...rest }) {
+    const { cellRenderer } = this.props
+
+    return cellRenderer({
+      ...rest,
+      columnIndex: columnIndex + columnIndex
+    })
+  }
+
+  _columnWidthRightGrid (index) {
+    const { fixedColumnCount, columnWidth } = this.props
+
+    return columnWidth instanceof Function
+      ? columnWidth(index + fixedColumnCount)
+      : columnWidth
+  }
+
+  _getBottomGridHeight () {
+    const { height } = this.props
+
+    let topGridHeight = this._getTopGridHeight()
+
+    return height - topGridHeight
+  }
+
+  _getLeftGridWidth () {
+    const { fixedColumnCount, columnWidth } = this.props
+
+    if (this._leftGridWidth == null) {
+      if (columnWidth instanceof Function) {
+        let leftGridWidth = 0
+
+        for (let index = 0; index < fixedColumnCount; index++) {
+          leftGridWidth += columnWidth(index)
+        }
+
+        this._leftGridWidth = leftGridWidth
+      } else {
+        this._leftGridWidth = columnWidth * fixedColumnCount
+      }
+    }
+
+    return this._leftGridWidth
+  }
+
+  _getRightGridWidth () {
+    const { width } = this.props
+
+    let leftGridWidth = this._getLeftGridWidth()
+
+    return width - leftGridWidth
```

Comment:
FYI I'm not actually able to reproduce the screenshot you've attached with scrollbars always visible. If I scroll all the way to the right it looks as I would expect:

<img width="205" alt="screen shot 2017-01-02 at 1 49 39 pm" src="https://cloud.githubusercontent.com/assets/29597/21596622/57656cf8-d0f2-11e6-83fc-4bb5cbff6df9.png">

<img width="214" alt="screen shot 2017-01-02 at 1 50 31 pm" src="https://cloud.githubusercontent.com/assets/29597/21596636/731309ba-d0f2-11e6-8935-90f107f41eb6.png">